packages/core/src/components/overlay/overlay.tsx

```diff
@@ -213,7 +220,9 @@ export class Overlay extends React.PureComponent<IOverlayProps, IOverlayState> {
                 {childrenWithTransitions}
             </TransitionGroup>
         );
-        if (usePortal) {
+        if (usePortal && container) {
+            return <Portal container={container}>{transitionGroup}</Portal>;
```

Comment:
❌ the prop should support being passed `undefined`, you should not need to do this extra check.
```suggestion
if (usePortal) { return <Portal container={container}>{transitionGroup}</Portal>; } else { return transitionGroup}
```
(neat new github feature!!)